scripts/svm/proposeRebalanceToHubPool.ts

```diff
@@ -0,0 +1,136 @@
+  const solanaCluster = evmChainId === CHAIN_IDs.MAINNET ? "mainnet" : "devnet";
+  const solanaChainId = getSolanaChainId(solanaCluster);
+  const svmUsdc = evmChainId === CHAIN_IDs.MAINNET ? SOLANA_USDC_MAINNET : SOLANA_USDC_DEVNET;
+
+  // Check there are no active proposals.
+  const currentRootBundleProposal = await hubPool.callStatic.rootBundleProposal();
+  if (currentRootBundleProposal.unclaimedPoolRebalanceLeafCount !== 0) {
+    throw new Error("Proposal has unclaimed leaves");
+  }
+
+  // Ensure bond token balance and approval is sufficient.
+  const bondTokenAddress = await hubPool.callStatic.bondToken();
+  const bondAmount = await hubPool.callStatic.bondAmount();
+  const bondToken = BondToken__factory.connect(bondTokenAddress, ethersProvider);
+  const bondBalance = await bondToken.callStatic.balanceOf(ethersSigner.address);
+  if (bondBalance.lt(bondAmount)) {
+    const ethDeposit = bondAmount.sub(bondBalance);
+    console.log(`Depositing ${ethers.utils.formatUnits(ethDeposit.toString())} ETH into bond token:`);
+    const tx = await bondToken.connect(ethersSigner).deposit({ value: ethDeposit });
+    console.log(`✅ submitted tx hash: ${tx.hash}`);
+    await tx.wait();
+    console.log("✅ tx confirmed");
+  }
+  const allowance = await bondToken.callStatic.allowance(ethersSigner.address, hubPool.address);
+  if (allowance.lt(bondAmount)) {
+    console.log(`Approving ${ethers.utils.formatUnits(bondAmount.toString())} bond tokens for HubPool:`);
+    const tx = await bondToken.connect(ethersSigner).approve(hubPool.address, bondAmount);
+    console.log(`✅ submitted tx hash: ${tx.hash}`);
+    await tx.wait();
+    console.log("✅ tx confirmed");
+  }
+
+  // Construct an empty pool rebalance tree as we need to propose at least one leaf.
+  const { poolRebalanceTree } = constructEmptyPoolRebalanceTree(solanaChainId, 0);
+  // Relayer refund root Merkle tree.
+  const { merkleTree } = constructSimpleRebalanceTreeToHubPool(netSendAmount, solanaChainId, new PublicKey(svmUsdc));
+
+  console.log("Proposing rebalance pool bundle to spoke...");
+  console.table([
+    { Property: "isTestnet", Value: process.env.TESTNET === "true" },
+    { Property: "originChainId", Value: evmChainId.toString() },
+    { Property: "targetChainId", Value: solanaChainId.toString() },
+    { Property: "hubPoolAddress", Value: hubPool.address },
+    { Property: "netSendAmount (formatted)", Value: formatUsdc(netSendAmount) },
+    { Property: "poolRebalanceRoot", Value: poolRebalanceTree.getHexRoot() },
+    { Property: "relayerRefundRoot", Value: merkleTree.getHexRoot() },
+  ]);
+
+  console.log("Submitting proposal...");
+  const tx = await hubPool.connect(ethersSigner).proposeRootBundle(
+    [0], // bundleEvaluationBlockNumbers, not checked in this script.
+    1, // poolRebalanceLeafCount, only one leaf in this script.
+    poolRebalanceTree.getHexRoot(), // poolRebalanceRoot.
```

Comment:
why is pool rebalance needed, cannot it be zero hash?

Reply:
I haven’t been able to do it with a zero hash in the poolRebalanceTree. I think we need to pass at least a tree with one leaf; the leaf can be constructed with zero hashes, except for the chain ID. Otherwise, we cannot get past this line:
https://github.com/across-protocol/contracts/blob/16060638d33df763436ec3cbc1336f11aba6be63/contracts/HubPool.sol#L635-L651

Additionally, the chain ID needs to be valid; otherwise, this line fails:
https://github.com/across-protocol/contracts/blob/16060638d33df763436ec3cbc1336f11aba6be63/contracts/HubPool.sol#L655

Reply:
oh, yes, you're totally right on this!